internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -34,62 +34,223 @@
+		Name:                   filepath.Join(m.cacheDir, "Objects", filePath),
+		Data:                   jsonData,
+		IsNoneMatchEtagEnabled: false,
+		EtagMatchConditions:    "",
+	})
+	if err != nil {
+		log.Debug("CreateFileFinalize :: Failed to put blob in storage: %v", err)
+	}
+	return err
 }
 
 // GetFile reads and returns the content of metadata for a file
 func (m *BlobMetadataManager) GetFile(filePath string) (*dcache.FileMetadata, error) {
-	// Dummy implementation
-	return nil, nil
+	// Get the metadata content from storage
+	data, err := m.storageCallbacks.GetBlobFromStorage(internal.ReadFileWithNameOptions{
+		Path: filepath.Join(m.cacheDir, "Objects", filePath),
+	})
+	if err != nil {
+		log.Debug("GetFile :: Failed to get metadata file content: %v", err)
+		return nil, err
+	}
+	// Unmarshal the JSON data into the Metadata struct
+	var metadata dcache.FileMetadata
+	err = json.Unmarshal(data, &metadata)
+	if err != nil {
+		log.Debug("GetFile :: Failed to unmarshal JSON data: %v", err)
+		return nil, err
+	}
+	// Return the Metadata struct
+	return &metadata, nil
 }
 
 // DeleteFile removes metadata for a file
 func (m *BlobMetadataManager) DeleteFile(filePath string) error {
-	// Dummy implementation
-	return nil
+	err := m.storageCallbacks.DeleteBlobInStorage(internal.DeleteFileOptions{
+		Name: filepath.Join(m.cacheDir, "Objects", filePath),
+	})
+	if err != nil {
+		if bloberror.HasCode(err, bloberror.BlobNotFound) {
+			log.Warn("DeleteFile :: DeleteBlobInStorage failed since blob is already deleted")
+			return nil
+		}
+	}
+	return err
 }
 
 // OpenFile increments the open count for a file and returns the updated count
 func (m *BlobMetadataManager) OpenFile(filePath string) (int64, error) {
-	// Dummy implementation
-	return 0, nil
+	count, err := m.updateHandleCount(filePath, true)
+	if err != nil {
+		log.Debug("OpenFile :: Failed to update file open count: %v", err)
+		return -1, err
+	}
+	return count, nil
 }
 
 // CloseFile decrements the open count for a file and returns the updated count
 func (m *BlobMetadataManager) CloseFile(filePath string) (int64, error) {
-	// Dummy implementation
-	return 0, nil
+	count, err := m.updateHandleCount(filePath, false)
+	if err != nil {
+		log.Debug("CloseFile :: Failed to update file close count: %v", err)
+		return -1, err
+	}
+	return count, nil
+}
+
+func (m *BlobMetadataManager) updateHandleCount(filePath string, increment bool) (int64, error) {
+	const maxBackoff = 30           // Maximum backoff time in seconds
+	backoff := 1 * time.Millisecond // Initial backoff time in milliseconds
+	var openCount int
+	for {
+		// Get the current handle count
+		attr, err := m.storageCallbacks.GetPropertiesFromStorage(internal.GetAttrOptions{
+			Name: filepath.Join(m.cacheDir, "Objects", filePath),
+		})
+		if err != nil {
+			log.Debug("updateHandleCount :: Failed to get handle count: %v", err)
+			return -1, err
+		}
+
+		if attr.Metadata["opencount"] != nil {
```

Comment:
since not having "opencount" metadata indicates some issue, we should bail out
Remove this check and add before this

```
// We never create file metadata blob w/o opencount property set.
if !attr.Metadata["opencount"] {
    log.Error("updateHandleCount :: File metadata blob found w/o opencount property: %s", filePath)
    return -1, <some appropriate error>
}

```